File: rabbitmq-jpa-kotlin/domain/src/main/code/sample/domain/model/CommentEntity.java
package sample.domain.model;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;


@Entity
@Table(name = "T_COMMENT")
@DynamicUpdate
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class CommentEntity extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    public CommentEntity(String content) {
        this.content = content;
    }

    @ManyToOne(optional = false)
    @JoinColumn(name = "article_id", insertable = true, updatable = false)
    private ArticleEntity article;

    @Column(length = 10, nullable = false)
    private String content;

//    @Column(nullable = false, insertable = true, updatable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
//    @CreatedDate
//    private LocalDateTime createdAt;
//
//    @Column(insertable = true, updatable = true, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
//    @CreatedDate
//    @LastModifiedDate
//    private LocalDateTime updatedAt;

    @Override
    public String toString() {
        return super.buildStringHelper()
            .add("article", article)
            .add("content", content)
            .toString();
    }
}

File: 2021-rabbitmq-jpa/domain/jpa/src/main/kotlin/sample/domain/jpa/model/comment/CommentEntity.kt
package sample.domain.jpa.model.comment

import org.hibernate.annotations.DynamicUpdate
import sample.domain.jpa.model.BaseEntity
import sample.domain.jpa.model.article.ArticleEntity
import javax.persistence.*

@Entity
@Table(name = "T_COMMENT")
@DynamicUpdate
class CommentEntity(
    article: ArticleEntity,
    @field:Column(
        length = 100,
        nullable = false
    )
    var content: String
) : BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long? = null

    @ManyToOne(optional = false)
    @JoinColumn(name = "article_id", insertable = true, updatable = false)
    private val article: ArticleEntity? = null
    override fun toString(): String {
        return super.buildStringHelper()
            .add("article", article)
            .add("content", content)
            .toString()
    }
}
